### PR TITLE
use github actions for CI checks

### PR DESCRIPTION
**Pull request by tariq1890**
This PR replaces TravisCI with GitHub Actions

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+name: Build
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (mrueg) [inline .github/workflows/ci.yml:7]: does that mean CI won't run on backporting PR to the release branches?

Comment (tariq1890) [reply .github/workflows/ci.yml:10]: I've fixed this. Thanks @mrueg !

**Comment by tariq1890** [comment]
@lilic 

> 
> I am not too sure about re-running Github Actions. I think that should be available to us from the GH interface. Not sure If that needs to be taken care of via the workflow YAML
> 
> Addressed. Created a trigger for tag pushes
> 
> I've parallelised it. You'll see a lot of duplication, unfortunately GitHub Actions does not re-usable steps as of yet. Once they do, we can remove the code duplication

**Comment by tariq1890** [comment]
Looks the GitHub Actions jobs (Benchmark and E2E) are faster than TravisCI by a minute. That's a win for us :)

**Comment by brancz** [comment]
Giving @lilic last call, but this lgtm! :tada: 

/lgtm
/hold

**Comment by lilic** [comment]
You seem to have a conflict, but otherwise lgtm, thanks Tariq!

**Pull request merged**